kclvm/parser/src/lexer/mod.rs

```
@@ -296,47 +296,100 @@ impl<'a> Lexer<'a> {
+                        token::CloseDelim(token::Bracket)
+                    }
+                    _ => token::CloseDelim(token::Paren),
+                },
+                None => {
+                    self.sess.struct_span_error_recovery(
                         "error nesting on close paren",
                         self.span(start, self.pos),
-                    )
+                    );
+                    token::CloseDelim(token::Paren)
                 }
-                self.indent_cxt.nesting -= 1;
-                token::CloseDelim(token::Paren)
-            }
+            },
             kclvm_lexer::TokenKind::OpenBrace => {
-                self.indent_cxt.nesting += 1;
+                self.indent_cxt.delims.push(token::OpenDelim(token::Brace));
                 token::OpenDelim(token::Brace)
             }
-            kclvm_lexer::TokenKind::CloseBrace => {
-                if self.indent_cxt.nesting == 0 {
-                    self.sess.struct_span_error(
+            kclvm_lexer::TokenKind::CloseBrace => match self.indent_cxt.delims.pop() {
+                Some(delim) => match delim {
+                    token::OpenDelim(token::Brace) => token::CloseDelim(token::Brace),
+                    token::OpenDelim(token::Paren) => {
+                        self.sess.struct_span_error_recovery(
+                            "error nesting on close brace",
+                            self.span(start, self.pos),
+                        );
+                        token::CloseDelim(token::Paren)
+                    }
+                    token::OpenDelim(token::Bracket) => {
+                        self.sess.struct_span_error_recovery(
+                            "error nesting on close brace",
+                            self.span(start, self.pos),
+                        );
+                        token::CloseDelim(token::Bracket)
+                    }
+                    _ => token::CloseDelim(token::Brace),
```

Pls using bug!() to raise an internal error instead of ignoring this error.